In [1]:
import numpy as np
import scipy.stats as stats
from scipy.stats import bartlett

**Задача:**<br>
Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: **173, 175, 180, 178, 177, 185, 183, 182**. Хоккеисты: **177, 179, 180, 188, 177, 172, 171, 184, 180**. Штангисты: **172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170**.

In [2]:
footb = np.array([173, 175, 180, 178, 177, 185, 183, 182])
hockey = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
w_lift = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

In [15]:
print (f'Sample size:\n football- {len(footb)}\n hockey-   {len(hockey)}\n w_lift-  {len(w_lift)}')

Sample size:
 football- 8
 hockey-   9
 w_lift-  11


In [24]:
def normality(n):
    print(stats.shapiro(n))
normality(footb)
normality(hockey)
normality(w_lift)

ShapiroResult(statistic=0.9775082468986511, pvalue=0.9495404362678528)
ShapiroResult(statistic=0.9579196572303772, pvalue=0.7763139009475708)
ShapiroResult(statistic=0.9386808276176453, pvalue=0.5051165223121643)


**Учитывая разную длину выборок и ненормальность распределений. Нужны непараметрические тесты.**

In [25]:
stats.kruskal(footb, hockey, w_lift)

KruskalResult(statistic=7.897493213863828, pvalue=0.01927885061595347)

**ОТВЕТ**: <u>учитывая, что pvalue равно 0.019 (1,9%), при alpha = 5% между группами наблюдаются значимые различия.</u>

<a href="https://github.com/PhYziCiAN/Stats.git" title="My GitHub">ссылка на GiHub</a>